src/libs/style.ts

```diff
@@ -74,7 +70,24 @@ function replaceSourceAccessToken(mapStyle: StyleSpecification, sourceName: stri
   if(!source) return mapStyle
   if(!("url" in source) || !source.url) return mapStyle
 
-  const accessToken = getAccessToken(sourceName, mapStyle, opts)
+  let authSourceName = sourceName
+  let setAccessToken = (url: string) => url.replace('{key}', accessToken)
```

Comment:
I would simplify this by defining a local variable called `url` or `sourceUrl` and use it instead of defining a local method which gets override in an if staement.

Reply:
I've refactored based on what I think you're suggesting here. For me it's just different and not simpler, but maybe I missed a simpler way to write it.

Reply:
Thanks!